### PR TITLE
RollbarAUL implementation

### PR DESCRIPTION
**Pull request by akornich**
## Description of the change

> New optional module for capturing relevant AUL entries with Rollbar SDK

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> n/a

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -104,6 +104,9 @@ let package = Package(
         .target(
             name: "RollbarAUL",
             dependencies: ["RollbarCommon", "RollbarNotifier"],
+            platforms: [
+                .macOS(.v10_15),
```
Comment (bishopb) [inline Package.swift:108]: Nit: any reason to exclude Big Sur?

**Review by bishopb — APPROVED** [review]

File: RollbarNotifier/Tests/RollbarNotifierTests-ObjC/RollbarTelemetryTests.m [hunk 2]
```diff
@@ -155,6 +155,7 @@ - (void)testRollbarLog {
     [Rollbar reapplyConfiguration];
 
     [RollbarTelemetry.sharedInstance clearAllData];
+    [NSThread sleepForTimeInterval:2.0f];
```
[the remark this answers is not in the document]
Comment (akornich) [reply RollbarNotifier/Tests/RollbarNotifierTests-ObjC/RollbarTelemetryTests.m:158]: this is just what seems to be a reasonable starting point. we do not want to keep the AUL monitoring thread too active. we can always adjust the interval in the future in either direction.

File: RollbarNotifier/Sources/RollbarNotifier/RollbarLogger.m [hunk 3]
```diff
@@ -261,7 +261,7 @@ - (void)log:(RollbarLevel)level
         return;
     }
     
-    RollbarConfig *config = self.configuration;
+    //RollbarConfig *config = self.configuration;
```
[the remark this answers is not in the document]
Comment (akornich) [reply RollbarNotifier/Sources/RollbarNotifier/RollbarLogger.m:264]: good point! gone!

File: Package.swift (hunk 1 above)
Comment (akornich) [reply Package.swift:108]: that means "starting from" version and excludes everything older than 10.15. So, the Big Sur is honored :)

File: RollbarNotifier/Sources/RollbarNotifier/RollbarConfigUtil.m [hunk 4]
```diff
@@ -110,14 +114,14 @@ + (BOOL)deleteFile:(nonnull NSString *)filePath
     if (fileExists) {
         BOOL success = [fileManager removeItemAtPath:filePath
                                                error:error];
-        if (!error) {
+        if (NO == success) {
+            RollbarSdkLog(@"Error while deleting file %@", filePath);
+        }
+        if ((NO == success) && (nil != error) && (nil != *error)) {
```
[the remark this answers is not in the document]
Comment (akornich) [reply RollbarNotifier/Sources/RollbarNotifier/RollbarConfigUtil.m]: agree. corrected. thank you!

**Comment by akornich** [comment]
@bishopb , thanks for the review and the approval!

File: RollbarAUL/Tests/RollbarAULTests-ObjC/RollbarAulStoreMonitorTest.m [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+    
+    NSBundle *mainBundle =
+    [NSBundle mainBundle];
+    
+    NSString *bundleIdentifier =
+    [mainBundle objectForInfoDictionaryKey:(NSString *)kCFBundleIdentifierKey];
+
+    os_log_t customLog = os_log_create(bundleIdentifier.UTF8String, "test_category");
+    int i = 0;
+    int maxIterations = 5;
```
[the remark this answers is not in the document]
Comment (akornich) [reply RollbarAUL/Tests/RollbarAULTests-ObjC/RollbarAulStoreMonitorTest.m:41]: nope, nothing magic - just generating a few test samples...

**Pull request merged**